src/Components/SourcesDropdown/index.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
+import { provisioningUrl } from '../../API/helpers';
+
+const SourcesDropdown = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [sources, setSources] = React.useState([]);
+  const onToggle = (isOpen) => {
+    fetchSources();
+    setIsOpen(isOpen);
+  };
+  const onFocus = () => {
+    const element = document.getElementById('toggle-secondary');
+    element.focus();
+  };
+  const onSelect = () => {
+    setIsOpen(false);
+    onFocus();
+  };
+
+  //TODO: move to react-query
+  const fetchSources = async () => {
+    const response = await fetch(provisioningUrl('sources'));
+    const json = await response.json();
+    setSources(json);
+  };
+
+  // TODO: add the chosen source to an global/external state
+  const sourcesDeatils = () => console.log(`${name} was chosen`);
+
+  const dropdownItemsMapper = () =>
+    sources.map(({ name, id }) => (
+      <DropdownItem onClick={sourcesDeatils} key={id}>
+        {name}
+      </DropdownItem>
+    ));
+  if (sources.length > 0) {
+    const items = dropdownItemsMapper();
```

Comment:
**NIT**: I think that this means we won't re-render in case sourcesDetails change (not needed to resolve now as there is no way to refetch the sources during the the life of the SourcesDropdown ATM anyway ^_^